packages/gatsby-theme-docs/src/components/code-sample-from-file.js

```diff
@@ -0,0 +1,24 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { CodeBlock } from '@commercetools-docs/ui-kit';
+
+// import Java from '../code-samples/test.java';
+
+// CodeBlockFromImport
+function CodeSampleFromFile(props) {
+  const [sample, setSample] = useState('');
+
+  // props.import - path to code sample
+  import(`../code-samples/${props.name}`).then(result => {
+    setSample(result.default);
+  });
+
+  return <CodeBlock content={sample} language={props.language} />;
+}
+
+CodeSampleFromFile.propTypes = {
+  name: PropTypes.string,
+  language: PropTypes.string,
```

Comment:
The language should be derived by the file name extension, we don't need to pass it explicitly.